ocaml/typing/ctype.ml

```diff
@@ -2032,82 +2032,103 @@ let rec estimate_type_jkind env ty =
   | Tpoly (ty, _) -> estimate_type_jkind env ty
   | Tpackage _ -> Jkind (value ~why:First_class_module)
 
-
-(* The ~fixed argument controls what effects this may have on `ty`.  If false,
-   then we will update the jkind of type variables to make the check true, if
-   possible.  If true, we won't (but will still instantiate sort variables).
-
-   The "fuel" argument here is used because we're duplicating the loop of
+(**** checking jkind relationships ****)
+
+type type_jkind_sub_result =
+  | Success
+    (* The [Type_var case might still be "success"; caller should check.
+       We don't just report success here because if the caller unifies the
+       tyvar, error messages improve. *)
+  | Type_var of Jkind.t * type_expr
+  | Missing_cmi of Jkind.t * Path.t
+  | Failure of Jkind.t
+
+(* The "fuel" argument here is used because we're duplicating the loop of
```

Comment:
nit: the `fuel` argument is no longer visible externally - might move this comment to the `loop` below.